Makefile

```diff
@@ -13,36 +13,48 @@
 # limitations under the License.
 
 # Old-skool build tools.
-# Simple makefile to build kind quickly and reproducibly in a container..
-# Only requires docker on the host
+# Simple makefile to build kind quickly and reproducibly
 #
 # Common uses:
 # - installing kind: `make install INSTALL_DIR=$HOME/go/bin`
 # - building: `make build`
 # - cleaning up and starting over: `make clean`
-
-# get the repo root and output path, go_container.sh respects these
+#
+################################################################################
+# ========================== Capture Environment ===============================
+# get the repo root and output path
 REPO_ROOT:=${CURDIR}
 OUT_DIR=$(REPO_ROOT)/bin
+# record the source commit in the binary, overridable
+COMMIT?=$(shell git rev-parse HEAD 2>/dev/null)
+################################################################################
+# ========================= Setup Go With Gimme ================================
+# go version to use for build etc.
+# setup correct go version with gimme
+PATH:=$(shell . hack/build/setup-go.sh && echo "$${PATH}")
+export PATH
+# work around broken PATH export
+SHELL:=env PATH=$(PATH) $(SHELL)
```

Comment:
this is the trickiest bit, at least on macOS default `make`, without this you fail to actually set path in trivial commands.

https://stackoverflow.com/questions/8941110/how-i-could-add-dir-to-path-in-makefile put me on the right track